### PR TITLE
Refactoring of the comments using STI [SCI-1090]

### PR DESCRIPTION
**Pull request by okriuchykhin**
https://biosistemika.atlassian.net/browse/SCI-1090

### CODE REVIEW

File: app/controllers/my_module_comments_controller.rb [hunk 1]
```diff
@@ -43,12 +43,14 @@ def index
...
+    @comment = TaskComment.new(
       message: comment_params[:message],
-      user: current_user)
+      user: current_user,
+      my_module: @my_module
+    )
 
     respond_to do |format|
-      if (@comment.valid? && @my_module.comments << @comment)
+      if @comment.valid? && @comment.save
```
Comment (Ducz0r) [inline app/controllers/my_module_comments_controller.rb]: I think we can replace this with only `if @comment.save`.

File: app/controllers/project_comments_controller.rb [hunk 2]
```diff
@@ -42,13 +42,14 @@ def index
...
       message: comment_params[:message],
-      user: current_user)
+      user: current_user,
+      project: @project
+    )
 
     respond_to do |format|
-
-      if (@comment.valid? && @project.comments << @comment)
+      if @comment.valid? && @comment.save
```
Comment (Ducz0r) [inline app/controllers/project_comments_controller.rb]: I think we can replace this with only `if @comment.save`.

File: app/controllers/result_comments_controller.rb [hunk 3]
```diff
@@ -40,12 +40,14 @@ def index
...
+    @comment = ResultComment.new(
       message: comment_params[:message],
-      user: current_user)
+      user: current_user,
+      result: @result
+    )
 
     respond_to do |format|
-      if (@comment.valid? && @result.comments << @comment)
+      if @comment.valid? && @comment.save
```
Comment (Ducz0r) [inline app/controllers/result_comments_controller.rb]: I think we can replace this with only `if @comment.save`.

File: app/controllers/step_comments_controller.rb [hunk 4]
```diff
@@ -39,13 +39,14 @@ def index
...
       message: comment_params[:message],
-      user: current_user)
+      user: current_user,
+      step: @step
+    )
 
     respond_to do |format|
-      if (@comment.valid? && @step.comments << @comment)
-
+      if @comment.valid? && @comment.save
```
Comment (Ducz0r) [inline app/controllers/step_comments_controller.rb]: I think we can replace this with only `if @comment.save`.

File: app/models/step.rb [hunk 5]
```diff
@@ -77,9 +75,8 @@ def self.search(user, include_archived, query = nil, page = 1)
   def destroy(current_user)
     @current_user = current_user
 
-    # Store IDs of comments, assets & tables so they
+    # Store IDs of assets & tables so they
     # can be destroyed in after_destroy
-    @c_ids = self.comments.collect { |c| c.id }
```
Comment (Ducz0r) [inline app/models/step.rb]: Just checking, this code (and the deletion afterwards) is no longer needed?

Comment (okriuchykhin) [inline app/models/step.rb]: Yeah, it gets deleted automatically with 'dependent: :destroy'

**Comment by okriuchykhin** [comment]
Unneeded validation calls removed

**Pull request merged**